index.js

```diff
@@ -1,45 +1,58 @@
-import express from "express";
 import geohash from "ngeohash";
 import {
   uniqueNamesGenerator,
   adjectives,
   colors,
   animals,
 } from "unique-names-generator";
+import express from "express";
+import { createServer } from "http";
+import { Server } from "socket.io";
+
+const app = express();
+const httpServer = createServer(app);
+const io = new Server(httpServer);
 
 /**
  * @type {Object.<string, Set>}
  */
 const grids = {};
-const app = express();
 
 app.use(express.json());
 app.use(express.static("public"));
 
-app.post("/api/initial", (req, res) => {
-  const { lat, lon } = req.body;
-  // geohash of length 5 is <= 4.89km X 4.89km
-  const hash = geohash.encode(lat, lon, 5);
+io.on("connection", (socket) => {
```

Comment:
In a subsequent PR, once the connection is established, the event handler will grab the user geohash from the handshake query and make the socket join a room with the name of the user's geohash. If their geohash is attr3, they'll join the room attr3. What do you think?

Reply:
We could even have a global event handler that handles when a user joins a room (geohash):

```js
io.of("/").adapter.on("join-room", (room, id) => {
  console.log(`socket ${id} has joined room ${room}`);
});
```

Reply:
Sounds good!